### PR TITLE
Fix issue with moving temporary files

### PR DESCRIPTION
**Pull request by mhidas**
Thanks @mphemming for identifying and fixing this issue.
(commits cherry-picked from #176)

### CODE REVIEW

**Comment by mphemming** [comment]
@mhidas happy for you to proceed with this one. I'll start a new branch for future pull requests ..

**Comment by mhidas** [comment]
@leonardolaiolo This is a quick & easy one to review.
The changes are actually from @mphemming but I can't merge it without an approving review, and GitHub doesn't let me request a review from him... 
(Let me know if you'd like more context)

**Review by leonardolaiolo — APPROVED** [review]

**Comment by leonardolaiolo** [comment]
Done @mhidas - I'll leave it to you to merge then.

**Pull request merged**
